api/types.go

```diff
@@ -115,3 +115,108 @@ type TaggedMetric struct {
 
 // GraphiteMetric is a flat, dot-separated identifier to a series of metric.
 type GraphiteMetric string
+
+// Column is a different aspect of data.
+// For example, Blueflood may stores (min / max / average / count) during rollups,
+// and these data are exposed via columns
+type Column string
+
+// Timerange represents a range of time a given time series is defined in:
+// it is 3-tuple of (start, end, resolution) with the following constraints:
+// start <= end
+// start = 0 mod resolution
+// end =   0 mod resolution
+type Timerange struct {
+	Start      int64
+	End        int64
+	Resolution int64
+}
+
+// IsValid determines whether the given timerange meets the constraint.
+func (tr Timerange) IsValid() bool {
+	return (tr.Start%tr.Resolution == 0 &&
+		tr.End%tr.Resolution == 0 &&
+		tr.Resolution > 0 &&
+		tr.Start <= tr.End)
+}
+
+// Slots represent the total # of data points
+func (tr Timerange) Slots() int {
+	return int((tr.Start - tr.End) / tr.Resolution)
+}
+
+// Timeseries is a single time series, identified with the associated tagset.
+type Timeseries struct {
+	Values []float64
+	TagSet TagSet
+}
+
+// SamplingStrategy determines how the given time series should be sampled.
+// Note(This is currently unused).
+type SamplingStrategy int
+
+const (
+	// SamplingMax chooses the maximum value.
+	SamplingMax SamplingStrategy = iota + 1
+	// SamplingMin chooses the minimum value.
+	SamplingMin
```

Comment:
I think by default it `iota` starts at 0 - i wanted to the values to start at 1 instead.